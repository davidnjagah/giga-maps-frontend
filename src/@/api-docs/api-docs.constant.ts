
export const ApiStatusColors = {
  approved: "#B9EFCB",
  declined: "#FFD7D9",
  initiated: "#FFE3C9",
  expired: "#FFD7D9"
} as Record<string, string>

export const ApiStatusName = {
  approved: "Active",
  declined: "Declined",
  initiated: "In Progress",
  expired: "Expired"
} as Record<string, string>


