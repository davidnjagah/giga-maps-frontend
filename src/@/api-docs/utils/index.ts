export * from './explore-api.utils';