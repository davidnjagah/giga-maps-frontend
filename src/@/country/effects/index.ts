// export { updateGlobalSchoolsFx } from './_update-global-schools-fx';
// export { updateSchoolsColorsFx } from './_update-schools-colors-fx';
// export { updateSchoolsFx } from './_update-schools-fx';
export { addCountriesFx } from './add-countries-fx';
export { zoomToCountryFx } from './zoom-to-country-fx';
