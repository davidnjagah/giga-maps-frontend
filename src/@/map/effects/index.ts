export { changeLayersFx, updateCoverageFilter } from './add-layers-fx'
export { changeStyleFx } from './change-style-fx';
export { initMapFx } from './init-map-fx';
