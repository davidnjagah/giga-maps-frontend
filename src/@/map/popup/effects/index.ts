export { updateSchoolPopupFx } from './update-school-popup.fx';
export { createLoadingPopupFx } from './create-school-popup-fx';
