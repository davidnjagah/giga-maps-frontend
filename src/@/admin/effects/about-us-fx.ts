import { createEffect } from "effector";

import { APIListType } from "~/api/types";
import { createRequestAuthFx } from "~/core/auth/effects/common.fx";

import { SaveFormListType } from "../models/about-us-model";
import { AboutUsImage } from "../types/about-us.type";

export const getImagesListFx = createEffect(() => {
  return createRequestAuthFx({
    url: `about_us/slide_image/?page_size=500&page=1`
  }) as Promise<APIListType<AboutUsImage>>
})


export const deleteImageFx = createEffect(({ body }: any) => {
  return createRequestAuthFx({
    method: 'DELETE',
    url: `about_us/slide_image/`,
    data: body
  }) as Promise<AboutUsImage>
})

export const uploadImagesFx = createEffect(({ formData }: { formData: any }) => {
  return createRequestAuthFx({
    method: 'POST',
    url: `about_us/slide_image/`,
    body: formData
  }) as Promise<AboutUsImage>
})

export const getAboutusContentFx = createEffect(() => {
  return createRequestAuthFx({
    url: `about_us/about_us/`,
  }) as Promise<SaveFormListType>
})

export const updateAboutusContentFx = createEffect(({ formData, isUpdate }: { formData: any, isUpdate: boolean }) => {
  return createRequestAuthFx({
    method: isUpdate ? 'PUT' : 'POST',
    url: `about_us/about_us/`,
    data: formData
  }) as Promise<SaveFormListType>
})