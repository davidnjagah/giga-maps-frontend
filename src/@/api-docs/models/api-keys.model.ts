import { combine, createEvent, createStore, sample } from "effector";

import { getApiKeyListFx, requestForApiKeyFx, requestForExtensionFx } from '~/@/api-docs/effects/api-keys-fx';
import { APIListType } from "~/api/types";
import { $loggedInUser } from "~/core/auth/models";
import { setPayload } from "~/lib/effector-kit";

import { ApiKeysType } from "../types/api-keys.type";

export const $apiKeysListResponse = createStore<null | APIListType<ApiKeysType>>(null);

$apiKeysListResponse.on(getApiKeyListFx.doneData, setPayload);

export const $apiKeysData = $apiKeysListResponse.map((resp) => resp?.results ?? []);
export const onRequestApiKey = createEvent<number>()
sample({
  clock: onRequestApiKey,
  fn: (api) => {
    return { api }
  },
  target: requestForApiKeyFx
})

// sample({
//   source: requestForApiKeyFx.failData,
//   fn: () => ({
//     title: 'The API is already available.',
//     kind: 'error',
//     subtitle: 'Please check the API keys section.'
//   }),
//   target: $notification
// })


sample({
  clock: requestForExtensionFx.done,
  source: combine({
    user: $loggedInUser
  }),
  fn: ({ user }) => {
    void getApiKeyListFx({ userId: user?.id })
  }
})

