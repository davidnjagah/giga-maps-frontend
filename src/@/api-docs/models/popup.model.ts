import { createEvent, restore, sample } from "effector";

import { requestForApiKeyFx } from "../effects/api-keys-fx";


export const onDownloadAPIPopup = createEvent<boolean>();
export const $downloadApiPopup = restore(onDownloadAPIPopup, false);

$downloadApiPopup.reset(requestForApiKeyFx.failData)

export const onDocumentAPIPopup = createEvent<boolean>();
export const $documentApiPopup = restore(onDocumentAPIPopup, false);
$documentApiPopup.reset(requestForApiKeyFx.failData)

export const onDownloadDataPopup = createEvent<boolean>();
export const $downloadDataPopup = restore(onDownloadDataPopup, false);

export const onSuccessMessagePopup = createEvent<boolean>();
export const $successMessagePopup = restore(onSuccessMessagePopup, false);

export const onRequestExtensionPopup = createEvent<number | null>();
export const $requestExtensionPopup = restore(onRequestExtensionPopup, null);

export const onRequestAPIPopup = createEvent<boolean>();
export const $requestAPIPopup = restore(onRequestAPIPopup, false);

sample({
  source: requestForApiKeyFx.doneData,
  fn: () => true,
  target: onSuccessMessagePopup
})

sample({
  source: requestForApiKeyFx.doneData,
  fn: () => false,
  target: $downloadApiPopup
})

sample({
  source: requestForApiKeyFx.doneData,
  fn: () => false,
  target: $documentApiPopup
})
