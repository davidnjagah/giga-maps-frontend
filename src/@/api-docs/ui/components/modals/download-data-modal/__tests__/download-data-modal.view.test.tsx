import { render } from '@testing-library/react'
import DownloadDataModalView from '../download-data-modal.view'
import { getExploreApiListFx } from '~/@/api-docs/effects/explore-api-fx';
import { apiList } from '~/tests/data/explore-apis-list';
import { onDownloadDataPopup } from '~/@/api-docs/models';
import { setCurrentExploreApi } from '~/@/api-docs/models/explore-api.model';


describe('DownloadDataModalView', () => {

  beforeEach(() => {
    fetchMock.mockResponse((req) => {
      if (req.url.includes('/accounts/apis')) {
        return Promise.resolve(JSON.stringify(apiList))
      } else {
        return Promise.resolve(JSON.stringify(null))
      }
    });

  })
  test('renders Modal component', async () => {
    await getExploreApiListFx();
    const { asFragment } = render(<DownloadDataModalView />)
    expect(asFragment).toMatchSnapshot();
  })

  test('renders Modal component with show popup', async () => {
    await getExploreApiListFx();
    setCurrentExploreApi(1);
    onDownloadDataPopup(true);
    const { asFragment } = render(<DownloadDataModalView />)
    expect(asFragment).toMatchSnapshot();
  })

  // test('renders ModalHeader component', () => {
  //   render(<DownloadDataModalView isOpen={true} onClose={() => { }} />)
  //   expect(screen.getByTestId('modal-header')).toBeInTheDocument()
  // })

  // test('renders ModalBody component', () => {
  //   render(<DownloadDataModalView isOpen={true} onClose={() => { }} />)
  //   expect(screen.getByTestId('modal-body')).toBeInTheDocument()
  // })

  // test('does not render when isOpen is false', () => {
  //   render(<DownloadDataModalView isOpen={false} onClose={() => { }} />)
  //   expect(screen.queryByTestId('modal')).not.toBeInTheDocument()
  // })
})
