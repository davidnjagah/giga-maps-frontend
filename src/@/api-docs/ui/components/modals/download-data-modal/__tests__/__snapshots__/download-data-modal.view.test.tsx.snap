// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadDataModalView renders Modal component 1`] = `[Function]`;

exports[`DownloadDataModalView renders Modal component with show popup 1`] = `[Function]`;
